development_bank_wales/pipeline/predictive_model/gaussian_smoothing.py

```diff
@@ -0,0 +1,102 @@
+# File: development_bank_wales/pipeline/predictive_model/gaussian_smoothing.py
+"""Gaussian smoothing for modelling competence."""
+
+# ---------------------------------------------------------------------------------
+
+import numpy as np
+import scipy
+
+# ---------------------------------------------------------------------------------
+
+
+def pcl_gaussian_smooth(dists, vals, sg_percentile=1.7):
+    """Gaussian smoothing for given features distances and values.
+
+    Args:
+        dists (np.array): ndarray of shape (n_pts, n_pts)
+            Squareform of all pairwise distances between points of the point
+            cloud, as computed e.g. by scipy.spatial.distance.pdist.
+        vals (np.array): ndarray of shape (n_pts, n_dims)
+             Values to be smoothed defined for each point. The different dimensions
+            are smoothed independently.
+        sg_percentile (float, optional): To determine the sigma (sg) of the Gaussian, the n-th percentile
+            of all distances in dists is used. This value specifies that n. Defaults to 1.7.
+
+    Returns:
+        smooth: ndarray of shape (n_pts, n_dims). Smoothed version of `vals`.
+    """
+
+    # Generate Gaussian function for smoothing
+    def gaussian_factory(mu, sg):
+        gaussian = (
+            lambda x: 1
+            / (sg * np.sqrt(2.0 * np.pi))
+            * np.exp(-1 / 2 * ((x - mu) / sg) ** 2.0)
+        )
+        return gaussian
+
+    gaussian_func = gaussian_factory(0.0, np.percentile(dists, sg_percentile))
+
+    # Smoothen the distances
+    gaud = gaussian_func(dists)
```

Comment:
Would the following work, so that we do  not have to define the gaussian probability density?
```suggestion
    # Smoothen the distances
    mu = 0.0
    sg = np.percentile(dists, sg_percentile)
    gaud = scipy.stats.norm.pdf(dists, mu, sg)
```

If not, can `gaussian_factory()` be defined outside of `pcl_gaussian_smooth()`?